crates/air_r_formatter/src/r/auxiliary/root.rs

```diff
@@ -23,4 +40,58 @@ impl FormatNodeRule<RRoot> for FormatRRoot {
             ]
         )
     }
+
+    fn is_suppressed(&self, node: &RRoot, f: &RFormatter) -> bool {
+        let Some(child) = node.expressions().first() else {
+            return false;
+        };
+
+        let comments = f.context().comments();
+
+        // Only consider very first comment in the file
+        let Some(comment) = comments.leading_comments(child.syntax()).first() else {
+            return false;
+        };
+
+        Self::is_suppression_comment(comment.piece().text())
+    }
+
+    // The default handling of suppressed nodes in `FormatNodeRule::fmt()`
+    // does not work out well for `RRoot` because it doesn't format
+    // trivia verbatim. So we use a custom implementation instead.
+    fn fmt_suppressed(&self, node: &RRoot, f: &mut RFormatter) -> FormatResult<()> {
+        let RRootFields { bom_token, .. } = node.as_fields();
+        write!(f, [bom_token.format()])?;
```

Comment:
IIUC you should not need this? You recursively format `node` and all its descendants, and since `node` is RRoot and the bom token is a descendant of that, it should get formatted too i think?